homeassistant/components/zwave_js/__init__.py

```diff
@@ -55,49 +62,70 @@ async def async_on_initialized():
         LOGGER.info("Connection to Zwave JS Server initialized")
         initialized.set()
 
-        # This should be moved to the Node Ready command or something
-        # And then listen to new nodes too.
-        dev_reg = await device_registry.async_get_registry(hass)
+        # run discovery on all ready nodes
         for node in client.driver.controller.nodes.values():
-            register_node_in_dev_reg(entry, dev_reg, client, node)
+            if not node.ready:
+                continue
+            asyncio.create_task(async_on_node_ready(node))
+        # TODO: register callback for "node_ready" event
```

Comment:
```suggestion
            if not node.ready:
                # TODO: register callback for "node_ready" event
                continue
            asyncio.create_task(async_on_node_ready(node))

        # TODO: register callback for "node added" controller event
```

Reply:
node added event is not that interesting as it will miss all/some info. node ready is better.
maybe only listen for node_added to give some feedback to the user that something is in the progress of being added ?